ex5B/ex5Spec.js

```diff
@@ -0,0 +1,53 @@
+/* jsl no-undef: "error" */
+/* global expect describe, it */
+//  http://stackoverflow.com/questions/16260779/how-to-disable-the-warning-define-is-not-defined-using-jshint-and-requirejs
+var firstUniqueChar = require('./ex5.js');
+
+describe('function firstUniqueChar', function () {
+  it('finds a unique character at start, one unique char', function () {
+    expect(firstUniqueChar('zabcdabcd')).toBe('z');
+  });
+
+  it('finds a unique character at middle, one unique char', function () {
+    expect(firstUniqueChar('abcdzabcd')).toBe('z');
+  });
+
+  it('finds a unique character at the end, one unique char', function () {
+    expect(firstUniqueChar('abcdabcdz')).toBe('z');
+  });
+
+  it('finds a unique character at start, 3 unique char', function () {
+    expect(firstUniqueChar('321abcdabcd')).toBe('3');
+  });
+
+  it('finds a unique character at middle, 3 unique char', function () {
+    expect(firstUniqueChar('abcd123abcd')).toBe('1');
+  });
+
+  it('finds a unique character at the end, 3 unique char', function () {
+    expect(firstUniqueChar('abcdabcd123')).toBe('1');
+  });
+
```

Comment:
Since we saw that the order of unique chars might be incorrectly calculated, add more tests:
`zabcd`, `abcdz`, `2b3c1a`, for example